### ISSUE TITLE
Changing of behaviour for attributeChanged

### ISSUE THREAD

**Issue by treshugart** [skatejs/skatejs#483]
The current behaviour is to not fire for existing attributes. See: https://github.com/skatejs/skatejs/blob/master/README.md#attribute-1. However, they've recently decided on some changes.

However, there have been several issues raised that are currently open that we should keep track of and action once they've been closed:

- https://github.com/w3c/webcomponents/issues/364 and https://github.com/w3c/webcomponents/issues/366
- https://github.com/w3c/webcomponents/issues/367

If this list grows, we should split these out but until it's final this may serve as a better aggregation point.

This is also dependent upon #511.
